[Sense HAT driver] Unable to read to I2C device

- Component affected:
  - [x] Distribution
    - Installer name: `kura_3.2.0_raspberry-pi-2-3-nn_installer.deb`
    - Platform: Raspberry Pi 3
    - Kura version affected: 3.2.0
    - Java version: `openjdk version "1.8.0_171"
OpenJDK Runtime Environment (build 1.8.0_171-8u171-b11-1~deb9u1-b11)
OpenJDK Client VM (build 25.171-b11, mixed mode)`
    - Host OS: Raspbian Stretch Lite
  - [x] Add-ons
    - Name: `org.eclipse.kura.example.driver.sensehat`
    - Version: 1.0.0
    - URL: https://marketplace.eclipse.org/content/sensehat-example-driver-eclipse-kura

- Textual description of the issue:

I am trying to use the **Sense HAT driver** to read data from the environmental sensors and send it to a Kapua instance. I installed the driver mentioned in the form but when I try to add a new driver I get some errors ([see kura.log](https://github.com/eclipse/kura/files/1988299/kura.log)) saying that it is not possible to read or write to the I2C device.
I have already enabled the I2C kernel module using the `raspi-config` tool and I confirm that the Sense HAT works because I am able to measure using the python library.

More details. I hope they are useful. First time I try to create a new driver after installing the sensehat package I get the following error:
<details><summary>kura.log</summary>
<p>

```
2018-05-10 07:32:02,509 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.e.d.s.SenseHatDriver - Activating SenseHat Driver...
2018-05-10 07:32:02,510 [Component Resolve Thread (Bundle 7)] INFO  o.e.k.e.d.s.SenseHatDriver - Opening Sense Hat...
2018-05-10 07:32:02,578 [Component Resolve Thread (Bundle 7)] ERROR o.e.k.r.s.s.LSM9DS1 - Could not create I2C Device
java.io.IOException: java.io.IOException
	at org.eclipse.kura.raspberrypi.sensehat.sensors.KuraI2CDevice.<init>(KuraI2CDevice.java:31)
	at org.eclipse.kura.raspberrypi.sensehat.sensors.LSM9DS1.getIMUSensor(LSM9DS1.java:190)
	at org.eclipse.kura.raspberrypi.sensehat.impl.SenseHatImpl.getIMUSensor(SenseHatImpl.java:68)
	at org.eclipse.kura.example.driver.sensehat.SenseHatInterface.<init>(SenseHatInterface.java:59)
	at org.eclipse.kura.example.driver.sensehat.SenseHatDriver.getSensehatInterface(SenseHatDriver.java:68)
	at org.eclipse.kura.example.driver.sensehat.SenseHatDriver.activate(SenseHatDriver.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146)
	at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620)
	at org.eclipse.equinox.internal.ds.ServiceReg.getService(ServiceReg.java:53)
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse$1.run(ServiceFactoryUse.java:212)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:210)
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111)
	at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:45)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:496)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:461)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:619)
	at org.osgi.util.tracker.ServiceTracker.addingService(ServiceTracker.java:414)
	at org.eclipse.kura.core.configuration.ConfigurationServiceImpl$3.addingService(ConfigurationServiceImpl.java:276)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:941)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:901)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464)
	at org.eclipse.equinox.internal.ds.InstanceProcess.registerService(InstanceProcess.java:536)
	at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:260)
	at org.eclipse.equinox.internal.ds.Resolver.buildNewlySatisfied(Resolver.java:473)
	at org.eclipse.equinox.internal.ds.Resolver.enableComponents(Resolver.java:217)
	at org.eclipse.equinox.internal.ds.SCRManager.processConfigurationEvent(SCRManager.java:403)
	at org.eclipse.equinox.internal.ds.SCRManager.processConfigurationEvent(SCRManager.java:303)
	at org.eclipse.equinox.internal.ds.WorkThread.run(WorkThread.java:91)
	at org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor.run(Executor.java:70)
Caused by: java.io.IOException
	at com.oracle.dio.i2cbus.impl.I2CSlaveImpl.open0(Native Method)
	at com.oracle.dio.i2cbus.impl.I2CSlaveImpl.<init>(I2CSlaveImpl.java:70)
	at com.oracle.dio.i2cbus.impl.I2CDeviceFactory.create(I2CDeviceFactory.java:43)
	at com.oracle.dio.i2cbus.impl.I2CDeviceFactory.create(I2CDeviceFactory.java:38)
	at jdk.dio.DeviceManager.openWithConfig(DeviceManager.java:290)
	at jdk.dio.DeviceManager.open(DeviceManager.java:273)
	at jdk.dio.DeviceManager.open(DeviceManager.java:209)
	at org.eclipse.kura.raspberrypi.sensehat.sensors.KuraI2CDevice.<init>(KuraI2CDevice.java:29)
	... 46 more
```

</p>
</details>

Then I tried some things.

1. I enable I2C interface using `raspi-config` tool -> Same exception as shown in the first comment.
2. I left `dtparam=i2c_arm=on` while commenting `i2c-dev` in `/etc/modules` -> Same exception
3. I commented `i2c-dev` from `/etc/modules` while keeping the line `dtparam=i2c_arm=on` in `/boot/config` -> java.io.IOException: java.io.IOException

So it looks like in a clean install, the raspberry pi 3 does not enable the I2C interface (this agrees with rasp documentation) and I get an IOException but if I load the kernel module, Kura sees the interface locked.
Let me know if you can replicate the issue.

Thanks!

--- a/_posts/devices/2018-04-12-sensehat-driver.md
+++ b/_posts/devices/2018-04-12-sensehat-driver.md
@@ -13,6 +13,11 @@ The driver allows access to the following resources:
 
 The driver-specific channel configuration contains a single parameter, **resource**, which allows to select the specific device resource that the channel is addressing (a sensor, a joystick event, etc).
 
+**Note about running on OpenJDK**
+
+If some exceptions reporting `Locked by other application` are visible in the log and the driver fails to start, try switching to the Oracle JVM by installing the `oracle-java8-jdk` package.
+For more information on the problem, please see [this](https://github.com/eclipse/kura/issues/2098) GitHub issue.
+
 ## Sensors
 
 The following values of the **resource** parameters refer to device sensors:

